images/virtualization-artifact/pkg/controller/vd/internal/source/sources.go

```diff
@@ -129,6 +129,25 @@ type CheckImportProcess interface {
 	CheckImportProcess(ctx context.Context, dv *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim) error
 }
 
+func setConditionFromStorageClassError(err error, condition *metav1.Condition) (bool, error) {
+	switch {
+	case err == nil:
+		return false, nil
+	case errors.Is(err, service.ErrDefaultStorageClassNotFound):
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = vdcondition.ProvisioningFailed
+		condition.Message = "Default StorageClass not found in the cluster: please provide a StorageClass name or set a default StorageClass."
+		return true, nil
+	case errors.Is(err, service.ErrStorageClassNotAvailable):
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = vdcondition.ProvisioningFailed
+		condition.Message = "Provided StorageClass not available."
```

Comment:
Actually, ErrStorageClassNotAvailable may be returned in 2 situations:
- StorageClass not provided (no sc in spec, in envs or no default cluster sc)
- Provided StorageClass is not allowed (either sc in spec, or cluster default sc).

The term "Available" is confusing in both situations.

Reply:
Fixed.

Reply:
Ok. I renamed `ErrStorageClassNotAvailable` -> `ErrStorageClassNotAllowed`. 
About `StorageClass not provided (no sc in spec, in envs or no default cluster sc)`: we already use error `ErrDefaultStorageClassNotFound` with message `Default StorageClass not found in the cluster: please provide a StorageClass name or set a default StorageClass.`